components/Tabs/src/Theming/MDCTabs+MaterialTheming.m

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCTabs+MaterialTheming.h"
+
+#import <Foundation/Foundation.h>
+
+#import <MaterialComponents/MaterialColorScheme.h>
+#import <MaterialComponents/MaterialTypographyScheme.h>
+
+static const CGFloat kUnselectedTitleOpacity = (CGFloat)0.6;
+static const CGFloat kUnselectedImageOpacity = (CGFloat)0.6;
+static const CGFloat kBottomDividerOpacity = (CGFloat)0.12;
+
+@implementation MDCTabBar (MaterialTheming)
+
+- (void)applyPrimaryThemeWithScheme:(nonnull id<MDCContainerScheming>)scheme {
+  id<MDCColorScheming> colorScheme = scheme.colorScheme;
+  [self applySemanticThemeWithColorScheme:colorScheme];
+
+  id<MDCTypographyScheming> typographyScheme = scheme.typographyScheme;
+  [self applyThemeWithTypographyScheme:typographyScheme];
+}
+
+- (void)applySurfaceVariantThemeWithScheme:(nonnull id<MDCContainerScheming>)scheme {
+  id<MDCColorScheming> colorScheme = scheme.colorScheme;
+  [self applySurfaceVariantThemeWithColorScheme:colorScheme];
+
+  id<MDCTypographyScheming> typographyScheme = scheme.typographyScheme;
+  [self applyThemeWithTypographyScheme:typographyScheme];
+}
+
+- (void)applySemanticThemeWithColorScheme:(id<MDCColorScheming>)colorScheme {
+  self.barTintColor = colorScheme.primaryColor;
+  self.tintColor = colorScheme.onPrimaryColor;
+  [self setTitleColor:colorScheme.onPrimaryColor forState:MDCTabBarItemStateSelected];
+  [self setImageTintColor:colorScheme.onPrimaryColor forState:MDCTabBarItemStateSelected];
+  UIColor *unselectedTitleColor =
+      [colorScheme.onPrimaryColor colorWithAlphaComponent:kUnselectedTitleOpacity];
+  UIColor *unselectedImageColor =
+      [colorScheme.onPrimaryColor colorWithAlphaComponent:kUnselectedImageOpacity];
+  [self setTitleColor:unselectedTitleColor forState:MDCTabBarItemStateNormal];
+  [self setImageTintColor:unselectedImageColor forState:MDCTabBarItemStateNormal];
+  self.bottomDividerColor =
+      [colorScheme.onPrimaryColor colorWithAlphaComponent:kBottomDividerOpacity];
```

Comment:
Should the divider always be applied for theming?  Looking at [the mocks in the Material Guidelines article for Tabs](https://material.io/design/components/tabs.html#usage), it doesn't appear as though they all have the divider enabled. I would expect that clients can apply the divider color **IF** they need the divider. However, in many cases (like including it in an AppBar), a divider is unnecessary and would not match the guidance in the article.

<img width="1156" alt="Screen Shot 2019-04-10 at 9 22 51 AM" src="https://user-images.githubusercontent.com/1753199/55882461-94103400-5b72-11e9-996e-92c9a46eebf9.png">

Reply:
I am not sure what's the best answer for this one. The current themer logic always applies the dividerColor. If this needs to be followed up on, I think a separate issue should be created on it.

Reply:
It's possible the Themer was written at a time when the divider was always applied. It seems at present that it's no longer set in all cases. I would suggest that we theme the component according to the latest guidance (since this is new code). We can debate the merits of changing the Themers in a separate Issue as you suggested.

Reply:
I think it's important to note that the Acceptance Criteria for this Issue never included matching the Themer behavior.  I've copied the criteria to the GitHub Issue for ease of reference.

Reply:
Addressed in 014c401360d496476332f1822b707265928fec51